integrationtest/backend/tck/src/main/java/org/hibernate/search/integrationtest/backend/tck/document/DocumentElementMultiValuedIT.java

```diff
@@ -0,0 +1,351 @@
+			level1.addValue( indexMapping.multiValuedFlattenedObject.multiValuedString, "1" );
+			level1.addValue( indexMapping.multiValuedFlattenedObject.multiValuedString, "2" );
+			level1.addValue( indexMapping.multiValuedFlattenedObject.multiValuedString, "3" );
+		} );
+	}
+
+	@Test
+	public void addValue_inSingleValuedNestedObject() {
+		expectSuccess( "1", document -> {
+			DocumentElement level1 = document.addObject( indexMapping.singleValuedNestedObject.self );
+			level1.addValue( indexMapping.singleValuedNestedObject.singleValuedString, "1" );
+		} );
+		expectSingleValuedException( "2", "singleValuedNestedObject.singleValuedString", document -> {
+			DocumentElement level1 = document.addObject( indexMapping.singleValuedNestedObject.self );
+			level1.addValue( indexMapping.singleValuedNestedObject.singleValuedString, "1" );
+			level1.addValue( indexMapping.singleValuedNestedObject.singleValuedString, "2" );
+		} );
+		expectSuccess( "3", document -> {
+			DocumentElement level1 = document.addObject( indexMapping.singleValuedNestedObject.self );
+			level1.addValue( indexMapping.singleValuedNestedObject.multiValuedString, "1" );
+		} );
+		expectSuccess( "4", document -> {
+			DocumentElement level1 = document.addObject( indexMapping.singleValuedNestedObject.self );
+			level1.addValue( indexMapping.singleValuedNestedObject.multiValuedString, "1" );
+			level1.addValue( indexMapping.singleValuedNestedObject.multiValuedString, "2" );
+			level1.addValue( indexMapping.singleValuedNestedObject.multiValuedString, "3" );
+		} );
+	}
+
+	@Test
+	public void addValue_inMultiValuedNestedObject() {
+		expectSuccess( "1", document -> {
+			DocumentElement level1 = document.addObject( indexMapping.multiValuedNestedObject.self );
+			level1.addValue( indexMapping.multiValuedNestedObject.singleValuedString, "1" );
+		} );
+		expectSingleValuedException( "2", "multiValuedNestedObject.singleValuedString", document -> {
+			DocumentElement level1 = document.addObject( indexMapping.multiValuedNestedObject.self );
+			level1.addValue( indexMapping.multiValuedNestedObject.singleValuedString, "1" );
+			level1.addValue( indexMapping.multiValuedNestedObject.singleValuedString, "2" );
+		} );
+		expectSuccess( "3", document -> {
+			DocumentElement level1 = document.addObject( indexMapping.multiValuedNestedObject.self );
+			level1.addValue( indexMapping.multiValuedNestedObject.multiValuedString, "1" );
+		} );
+		expectSuccess( "4", document -> {
+			DocumentElement level1 = document.addObject( indexMapping.multiValuedNestedObject.self );
+			level1.addValue( indexMapping.multiValuedNestedObject.multiValuedString, "1" );
+			level1.addValue( indexMapping.multiValuedNestedObject.multiValuedString, "2" );
+			level1.addValue( indexMapping.multiValuedNestedObject.multiValuedString, "3" );
+		} );
+	}
+
+	@Test
+	public void inheritedFromFlattenedParent() {
+		// -> multi-valued flattened => children are multi (whether explicit or not)
```

Comment:
is it supposed to be implemented?

Reply:
Sorry, I was really messy this time... This was just something I wrote initially to sum up what needs to be tested. It's obsolete, I removed it.